### PR TITLE
Object respawn functions

### PR DESCRIPTION
**Pull request by FileEX**
Added ``respawnObject`` and ``toggleObjectRespawn`` to the server-side.
Added new function ``isObjectRespawnable`` as shared function.

This PR is part of #3405

### CODE REVIEW

**Comment by TracerDS** [comment]
Would `toggleObjectRespawn` work only if the object broke?

**Comment by FileEX** [comment]
Yes - I don't think it matters when it comes to returning false for toggleObjectRespawn, because the object respawn logic checks if the object has 0 hp anyway, so returning false for unbreakable objects won't change much here

https://github.com/multitheftauto/mtasa-blue/blob/d93287de761e568400b3b555a277e4ead6546ca3/Client/mods/deathmatch/logic/CClientObject.cpp#L432-L434

https://github.com/multitheftauto/mtasa-blue/blob/d93287de761e568400b3b555a277e4ead6546ca3/Client/mods/deathmatch/logic/CClientObject.cpp#L463-L471

File: Server/mods/deathmatch/logic/CObject.h [hunk 1]
```diff
@@ -81,6 +81,9 @@ class CObject : public CElement
     bool IsBreakable() { return m_bBreakable; }
     void SetBreakable(bool bBreakable) { m_bBreakable = bBreakable; }
 
+    bool IsRespawnEnabled() const { return m_bRespawnable; }
+    void SetRespawnEnabled(bool bRespawn) { m_bRespawnable = bRespawn; }
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CObject.h]: Make it `noexcept`, please

File: Server/mods/deathmatch/logic/luadefs/CLuaObjectDefs.cpp [hunk 2]
```diff
@@ -312,3 +318,18 @@ bool CLuaObjectDefs::BreakObject(CObject* const pObject)
...
+
+bool CLuaObjectDefs::ToggleObjectRespawn(CObject* const pObject, const bool bRespawn)
+{
+    return CStaticFunctionDefinitions::ToggleObjectRespawn(pObject, bRespawn);
+}
+
+bool CLuaObjectDefs::IsObjectRespawnable(CObject* const pObject)
+{
+    return pObject->IsRespawnEnabled();
+}
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/luadefs/CLuaObjectDefs.cpp]: Make it `noexcept`, please

**Review by TracerDS — APPROVED** [review]
Good job!

File: Client/mods/deathmatch/logic/luadefs/CLuaObjectDefs.cpp [hunk 3]
```diff
@@ -707,3 +709,15 @@ int CLuaObjectDefs::SetObjectProperty(lua_State* luaVM)
     lua_pushboolean(luaVM, false);
     return 1;
 }
+
+bool CLuaObjectDefs::IsObjectRespawnable(CClientEntity* const pEntity)
```
Comment (TracerDS) [inline Client/mods/deathmatch/logic/luadefs/CLuaObjectDefs.cpp]: `noexcept`?

File: Client/mods/deathmatch/logic/luadefs/CLuaObjectDefs.h [hunk 4]
```diff
@@ -29,6 +29,7 @@ class CLuaObjectDefs : public CLuaDefs
     LUA_DECLARE(GetObjectMass);
     LUA_DECLARE(GetObjectProperty);
     LUA_DECLARE(GetObjectProperties);
+    static bool IsObjectRespawnable(CClientEntity* const pEntity);
```
Comment (TracerDS) [inline Client/mods/deathmatch/logic/luadefs/CLuaObjectDefs.h]: `noexcept`?

File: Client/mods/deathmatch/logic/rpc/CObjectRPCs.cpp [hunk 5]
```diff
@@ -130,8 +132,24 @@ void CObjectRPCs::SetObjectBreakable(CClientEntity* pSource, NetBitStreamInterfa
...
+
+void CObjectRPCs::RespawnObject(CClientEntity* pSource, NetBitStreamInterface& bitStream)
+{
+    auto* pObject = static_cast<CDeathmatchObject*>(m_pObjectManager->Get(pSource->GetID()));
+
+    if (pObject)
+        g_pClientGame->GetObjectRespawner()->Respawn(pObject);
+}
+
+void CObjectRPCs::ToggleObjectRespawn(CClientEntity* pSource, NetBitStreamInterface& bitStream)
```
Comment (TracerDS) [inline Client/mods/deathmatch/logic/rpc/CObjectRPCs.cpp:149]: `noexcept`?

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp [hunk 6]
```diff
@@ -8347,6 +8347,43 @@ bool CStaticFunctionDefinitions::SetObjectBreakable(CElement* pElement, const bo
...
+    if (!pObject)
+        return false;
+
+    pObject->SetRespawnEnabled(bRespawn);
+
+    CBitStream BitStream;
+    BitStream->WriteBit(bRespawn);
+    m_pPlayerManager->BroadcastOnlyJoined(CElementRPCPacket(pObject, TOGGLE_OBJECT_RESPAWN, *BitStream.pBitStream));
+
+    return true;
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp]: `noexcept`?

File: Server/mods/deathmatch/logic/CStaticFunctionDefinitions.h [hunk 7]
```diff
@@ -422,6 +422,8 @@ class CStaticFunctionDefinitions
     static bool BreakObject(CElement* pElement);
     static bool SetObjectVisibleInAllDimensions(CElement* pElement, bool bVisible, unsigned short usNewDimension = 0);
     static bool SetObjectBreakable(CElement* pElement, const bool bBreakable);
+    static bool RespawnObject(CElement* const pElement);
+    static bool ToggleObjectRespawn(CElement* const pElement, const bool bRespawn);
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CStaticFunctionDefinitions.h]: `noexcept`?

File: Server/mods/deathmatch/logic/luadefs/CLuaObjectDefs.cpp [hunk 8]
```diff
@@ -312,3 +318,18 @@ bool CLuaObjectDefs::BreakObject(CObject* const pObject)
...
+{
+    return CStaticFunctionDefinitions::RespawnObject(pObject);
+}
+
+bool CLuaObjectDefs::ToggleObjectRespawn(CObject* const pObject, const bool bRespawn)
+{
+    return CStaticFunctionDefinitions::ToggleObjectRespawn(pObject, bRespawn);
+}
+
+bool CLuaObjectDefs::IsObjectRespawnable(CObject* const pObject) noexcept
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/luadefs/CLuaObjectDefs.cpp]: `noexcept`?

**Review by TracerDS — APPROVED** [review]

**Review by tederis — APPROVED** [review]
LGTM

**Pull request merged**